### PR TITLE
Pull request for Issue #113 new

### PR DESCRIPTION
**Pull request by matevzmihalic**
This is new pull request. It has everything that we discussed in previous request, but is now synced with master.
I removed image sizes, because the only common size that we can get from all networks is 50x50. If we want other sizes, we should resize it ourself.

### CODE REVIEW

File: piplmesh/settings.py [hunk 1]
```diff
@@ -278,3 +278,5 @@
 # you will be explicitly warned that you have to change the code to take effect, before you will make the change.
 # Current settings are autocomplete, searching whole web.
 SEARCH_ENGINE_UNIQUE_ID = '003912915932446183218:zeq20qye9oa'
+
+DEFAULT_IMAGE_PATH = 'piplmesh/images/unknown.png'
```
Comment (mitar) [inline piplmesh/settings.py]: Good!

`DEFAULT_IMAGE_PATH` -> `DEFAULT_USER_IMAGE_PATH`

File: piplmesh/frontend/templates/user_image.html [hunk 2]
```diff
@@ -0,0 +1,2 @@
+{% load i18n %}
+<img src="{{ user_image.url }}" class="image_{{ user_image.class }}" alt="{% trans "User picture" %}" />
```
Comment (mitar) [inline piplmesh/frontend/templates/user_image.html]: What we will have? _User picture_? Or _User image_?

Comment (mitar) [inline piplmesh/frontend/templates/user_image.html]: (I vote for image.)

File: piplmesh/account/utils.py [hunk 3]
```diff
@@ -1,17 +1,13 @@
...
     Format Facebook Graph API URL. 
     """
     
     param = ''
-    if user:
-        profile = models.UserProfile.objects.get(user=user)
     if user and token:
-        param = '?access_token=%s' % profile.token
+        param = '?access_token=%s' % user.facebook_token
     results = 'https://graph.facebook.com/%s/%s' % (self, param)
```
Comment (mitar) [inline piplmesh/account/utils.py]: ??? `self` passed to `%s`? We depend on how object is stringified? Ugly!

File: piplmesh/account/templatetags/facebook.py [hunk 4]
```diff
@@ -17,19 +17,6 @@ def facebook_graph(self):
```
Comment (mitar) [inline piplmesh/account/templatetags/facebook.py:-1]: Are those functions used anywhere? I don't think there is a reason for them to be in the templates anyway. This should go somewhere else (to Facebook API code or to User class methods).

File: piplmesh/account/templatetags/account.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+from django.contrib.staticfiles.storage import staticfiles_storage
+from django.conf import settings
+
+from piplmesh.account import models
+
+register = template.Library()
+
+@register.inclusion_tag('user_image.html', takes_context=True)
+def user_image(context, user=None):
+    if user==None:
```
Comment (mitar) [inline piplmesh/account/templatetags/account.py]: `user is None`.

File: piplmesh/account/templatetags/account.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+        user = context['user']
+    
+    return {
+        'user_image': {
+            'url': user.image_url(context['request'].build_absolute_uri(staticfiles_storage.url(settings.DEFAULT_IMAGE_PATH))),
```
Comment (mitar) [inline piplmesh/account/templatetags/account.py]: Oh, just pass request to `image_url` and do the rest there.

File: piplmesh/account/models.py [hunk 7]
```diff
@@ -46,3 +48,18 @@ class User(auth.User):
     connections = mongoengine.ListField(mongoengine.EmbeddedDocumentField(Connection))
     connection_last_unsubscribe = mongoengine.DateTimeField()
     is_online = mongoengine.BooleanField(default=False)
+    
+    def image_url(self, default_url=staticfiles_storage.url(settings.DEFAULT_IMAGE_PATH)):
```
Comment (mitar) [inline piplmesh/account/models.py]: Rename to `get_image_url`.

File: piplmesh/account/templatetags/account.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+            'class': '50_50'
```
Comment (mitar) [inline piplmesh/account/templatetags/account.py]: Then remove this class if we cannot set size.

File: piplmesh/frontend/templates/user_image.html (hunk 2 above)
Comment (mitar) [inline piplmesh/frontend/templates/user_image.html]: Just name CSS class `user_image`.

File: piplmesh/account/templatetags/account.py (hunk 5 above)
Comment (matevzmihalic) [inline piplmesh/account/templatetags/account.py]: It is by default, but you could pass user to this tag and get it's image.

Comment (mitar) [inline piplmesh/account/templatetags/account.py]: I know what the code does. I am telling that you should use `user is None` instead of `user==None`.

**Comment by mitar** [comment]
Please please please finalize this. Because there are more and more hacks elsewhere because this code is missing.

Update from main repository.

**Comment by matevzmihalic** [comment]
All done. I hope I merged it correctly :)

**Comment by mitar** [comment]
You mean issue #63.

**Comment by mitar** [comment]
There are some things to fix, but I will merge it anyway, because it is needed.

**Comment by mitar** [comment]
Please see above changes to see what should be done.

**Pull request merged**